File: LeetCode_Solutions/1044-find-common-characters/find-common-characters.java
class Solution {
    public List<String> commonChars(String[] words) {
        int[] lcnt=new int[26];
        Arrays.fill(lcnt,Integer.MAX_VALUE);
        for(String str:words){
            int[] currentLetter=new int[26];
            for(int i=0;i<str.length();i++){
                currentLetter[str.charAt(i)-'a']++;
            }
            for(int i=0;i<26;i++){
                lcnt[i]=Math.min(lcnt[i],currentLetter[i]);
        }
        }
        List<String> ls=new ArrayList<>();
        for(int i=0;i<26;i++){
            while(lcnt[i]>0){
                ls.add(String.valueOf((char)(i+'a')));
                lcnt[i]--;
            }
        }
        return ls;
    }
}

File: LeetCode_Solutions/1044-find-common-characters/find-common-characters.py
class Solution:
    def commonChars(self, words: List[str]) -> List[str]:
        lst=[2**31 for _ in range(26)]
        for str1 in words:
            cnt=Counter(str1)
            for i in range(26):
                lst[i]=min(lst[i],cnt[chr(i+ord('a'))])
        res=[]
        for i in range(26):
            while lst[i]>0:
                res.append(chr(i+ord('a')))
                lst[i]-=1
        return res